external/model-preparation-algorithm/mpa_tasks/extensions/datasets/mpa_cls_dataset.py

```diff
@@ -9,26 +9,24 @@
 from mmcls.datasets.builder import DATASETS, PIPELINES
 from mmcls.datasets.pipelines import Compose
 from mmcls.datasets.base_dataset import BaseDataset
+from mpa_tasks.utils.data_utils import get_cls_img_indices
 from mpa.utils.logger import get_logger
 
 logger = get_logger()
-
-
 @DATASETS.register_module()
 class MPAClsDataset(BaseDataset):
 
-    def __init__(self, old_new_indices=None, ote_dataset=None, labels=None, **kwargs):
+    def __init__(self, ote_dataset=None, labels=None, **kwargs):
         self.ote_dataset = ote_dataset
         self.labels = labels
         self.CLASSES = list(label.name for label in labels)
         self.gt_labels = []
         pipeline = kwargs['pipeline']
-        self.img_indices = dict(old=[], new=[])
         self.num_classes = len(self.CLASSES)
 
-        if old_new_indices is not None:
-            self.img_indices['old'] = old_new_indices['old']
-            self.img_indices['new'] = old_new_indices['new']
+        test_mode = kwargs.get('test_mode', False)
```

Comment:
Do we need this option?

Reply:
Validation set also uses this MPA***dataset. There's no need to seperate indices in case of validation. So I added this option.